src/arrayMethodJoin.js

```diff
@@ -4,8 +4,28 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        result += ',';
+        continue;
+      }
```

Comment:
```suggestion
      if (this[i] !== null || this[i] !== undefined) {
        result += this[i]
      }
```
You may always add `this[i]` if `this[i] !== null/undefined`. Othervise nothing will be added